### DISCUSSION TITLE
Is there an expiry time on restoring a deleted branch in a merged PR?

### DISCUSSION THREAD

**Discussion by benjaminhwilliams** [community/community#23262] [New to GitHub]
<p>I have a collaborator who is nervous about the longevity of the ‘restore’ functionality of deleted branches in merged PRs.  Can anyone provide some clarity as to how long a merged and deleted PR branch is retained for restoring?</p>
<p>I was under the impression that one can restore a deleted PR branch indefinitely, but I can’t find confirmation of that on the GitHub support pages.  The only information is second-hand, a quote from a GitHub helpdesk reply from 2013 in <a href="https://stackoverflow.com/a/17954767" rel="nofollow noopener">this StackOverflow answer</a>:</p>
<blockquote>
<p>We use a separate ref namespace for all Pull Requests which we use for various things including restoring the branch. Since we keep those [Pull Request] refs indefinitely, <strong><em>there’s no time limit on restoring a branch</em></strong>.</p>
</blockquote>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>I don’t have a very satisfying answer for you I’m afraid. The branches will be restorable indefinitely, <em>unless</em> the repository cache is cleared.</p>
<p>We don’t clear repository caches automatically (we’re in the version control business, so we don’t delete data unless we absolutely have to :wink:) so usually the only reason we would do that is if we had someone writing in to us asking for us to clear them as <a href="https://help.github.com/articles/remove-sensitive-data/" rel="nofollow noopener">part of the sensitive data removal process</a> (Step 8).</p>
<p>It’s also possible we might clear the cache for technical reasons, if the content or structure of the repo was causing us difficulties to host it, but that would usually only happen if the repo was exceptionally large or had a wildly structured folder layout.</p>
<p>Sorry I don’t have a better answer for you!</p>

**Reply by activenode** [reply]
Isn't that a bit cumbersome? Like human failure happens, right? Now what if you had something as a branch you pushed accidentally and now there's now way for other people in your team to prevent removing it? 

That's like saying: "Well better be careful because if you aren't then you might be screwed"

Which is all but humane IMO.

**Reply by activenode** [reply]
PS: I'm not talking about merged branches. Well what I could do - now thinking about it - is restoring it, push force an empty commit.

**Comment by benjaminhwilliams** [comment]
<p>Thanks <a class="mention" href="/that-pat">@that-pat</a>, that’s very helpful indeed.</p>
